### ISSUE TITLE
Io port not inserting items to chest

### ISSUE THREAD

**Issue by HEROgold** [bitwiseshiftleft/factorio-router#2]
I got a red wire connected to an IO terminal IO port. This signal holds -100 iron plate.
The other IO Terminal then sends the iron plate from the connected chest, into the router network, and when it arrives at the IO terminal thats requesting the items, its not putting it into the connected chest.

**Comment by bitwiseshiftleft** [comment]
Could you post a screenshot?

By design, I/O requests should be positive, and connected to the terminal body (not the I/O pole) by a green wire.  The documentation should be better on this, sorry (it's on GitHub but not yet on the mod page, or in any kind of in-game help).  Your way might actually work to, but I haven't tested it.  But anyway, the hidden inserters that load the chest are not filter inserters, so they should put them in the chest regardless.  Or are the opposite inserters instead returning the plates to the network?

There is also a known bug with I/O terminals that could cause this, which is in some sense a Factorio bug that I haven't done the work to get around yet: if you hold a belt and drag it over the I/O hub, the Factorio belt-dragging code can reverse the belt component of the I/O hub (which is actually a zero-length underneathie, so that stuff can't flow past the hub by accident).  This happens even though the hub's underneathies are marked as non-rotatable, and it doesn't fire a particularly useful Lua event when it happens.  If this happened in your network, you can either rebuild the hub, or drag a belt in the opposite direction to flip it back.

**Comment by HEROgold** [comment]
![image](https://github.com/bitwiseshiftleft/factorio-router/assets/21345384/b8c0fd4b-44d6-4e73-828e-45463b8fd2db)

Even when i am manually dropping items on the belt (as seen on the right side, the wooden boxes) the item's are not being pushed into the chest

(ignore the upgrade symbol)
To confirm, this screenshot shows the belt under the Terminal point into the chest
![image](https://github.com/bitwiseshiftleft/factorio-router/assets/21345384/f602645c-488f-4cf5-81ce-f013612757ea)
Where as rotating that belt using the drag over feature, it points outwards
![image](https://github.com/bitwiseshiftleft/factorio-router/assets/21345384/59f1928c-6797-4bbe-9f78-31da988a1a2f)

When testing i made sure the belt of the terminal points towards the chest, and has items flowing into it

**Comment by HEROgold** [comment]
Even on the left side, no power poles are being pushed to the chest, they are AAI Strongboxes, if that matters

**Comment by bitwiseshiftleft** [comment]
Hm.  Could you give me a blueprint (if it reproduces when blueprinted) or a save file?  A similar build works in my test setup.

**Comment by HEROgold** [comment]
Here's the blueprint string, I copied, pasted tested (same results) and then blueprinted it.

0eNrlV9GOoyAU/ReetRG0Kv7KpmlQaUsWwSBOpmn8gP2L/bb9kgXNdNoOdLSdh83sS1MRzr33cDhXTqDkPW0VExoUJ8AqKTpQ/DiBju0F4XZMH1sKCsA0bUAABGnsk5K9pipMXpOwa4jSYAgAEzV9BQUcgk+X70inQ62I6FqpdFhSfomAhk0AqNBMMzplMz4ct6JvSqpMiDOQzVcTocNKNiUTREtlorSyM2ulsPEtHsSrdQCOZqH5Y+LUTNFqmpAGFkMrybclPZAXZgDMqh3jpr5ZVJj6OQ+ZkiKk3MAqVoWt5NQWVMneEguj6IKezfhCiCmFzgJD+7NXlIrLclkNithMHyyjNxSgc/yG1qxvbmJ/5ABFbxzkIweCsv2hlP1YZLpxhIhdu83kPWwEr9lFCyqFnkoTv+gcmZzzcCCtb5GMaFopzKwJLxzFyEnT3kGOVjBNohStB7dyKqaqnumteVefAXZMdXr7QUgvTOnejNzmNU0MORM/rYo6asGWL6dkN4mwaYkaj0YB/vz6bcf6jpoMubS7r1VPXXSliyWG70gMuSSWzZcY9kgsdcDmZ1hCWNiZPRL7Ur46neEd1KVUResPOvUJFd+1N0dJcHXW1PrzomD0MD68xUcufOg/HkyG7+x4T8iVC8xkM7de7/aDtYdniPyJjnwYjTb3G8JFqqsUYwTzzHee73aCT06gPijaHSSvr3rBVStw1Rcv3Wi0bKOTh/HnCfU5n03eDPZKS9/FZ5c0RJ/RwPS5RrbK4zWOs/+eYIR8BGdf8aUA33j+9l8KMF/qKNkyx8IP489yLLS4teaL8kfwYfx5+aNnWzd+oHWjxN+7U9+1JX6+d+N3bhBO/6XOjZLZH54xXM73ZFeG8/ECWlxc3QPwYiqd9JfDJMMoy1McxVEyDH8BzDFigg==

**Comment by bitwiseshiftleft** [comment]
Huh, this works for me.  Some of the poles get stuck under the router (because the box fills and there is no longer demand for them) but nothing gets stuck in the I/O port.  What mods are you using?

**Comment by HEROgold** [comment]
Here is my world file
https://drive.google.com/file/d/1PDcd0YYqcQiSaKwHcYL2Y3nx2rnjeVSS/view?usp=drive_link
Hope that helps

**Comment by bitwiseshiftleft** [comment]
Give it a try with 0.1.3 (requires rebuilding your routers [Edit: I mean, I/O terminals]).  I believe the issue comes from a poor interaction with FilterInsertersBegone, which I have attempted to work around.

Basically this mod creates hidden inserters, both filter and non-filter.  FilterInsertersBegone changes the prototype so they're all filter.  That breaks the non-filter ones (inserting into the I/O terminals), because they're in whitelist mode by default and have no filters set.  FilterInsertersBegone gets a notification on build, and considers changing them to blacklist, but then it doesn't change them because they're already connected to the circuit network.  So they stay broken.

I've patched my mod to detect that its supposedly "non-filter" inserters are actually filtered, and set them to blacklist.  This happens when you build the I/O terminal, so new ones should work, hopefully.  I haven't tested it with super-fast belts but hopefully it still works there.
